### PR TITLE
Add policy migration

### PR DESCRIPTION
**Pull request by swiftgist**
Signed-off-by: Eric Jackson <ejackson@suse.com>

### CODE REVIEW

File: srv/modules/runners/push.py [hunk 1]
```diff
@@ -132,6 +152,56 @@ def output(self, common):
...
+        """
+        Migrate the original data structure to the ceph namespace data
+        structure
+        """
+        if 'storage' in yaml and 'osds' in yaml['storage']:
+            yaml['ceph'] = {}
+            yaml['ceph']['storage'] = {}
+            yaml['ceph']['storage']['osds'] = {}
+            for osd in yaml['storage']['osds']:
+                yaml['ceph']['storage']['osds'][osd] = { 'format': 'bluestore' }
```
Comment (jschmid1) [inline srv/modules/runners/push.py]: Although most of the time correct, we change the format to bluestore regardless of the user's choice. Same with the wal/db distrubution to the old journal.

File: srv/modules/runners/push.py [hunk 2]
```diff
@@ -132,6 +152,56 @@ def output(self, common):
...
+    def rename(self, filename):
+        """
+        Copy the policy.cfg and update the existing file
+        """
+        shutil.copyfile(filename, "{}-original".format(filename))
+        lines = []
+        with open(filename, "r") as policy:
+            for line in policy:
+                if line.startswith('profile-'):
+                    line = "migrated-{}".format(line)
```
Comment (jschmid1) [inline srv/modules/runners/push.py]: That would end up as:

`migrated-profile-default/` right?

A bit confusion when you were always used to scan for profile-*. Maybe change it to `profile-XXXX-migrated` ?

**Comment by jschmid1** [comment]
code works as intended with storage:osds and storage:osd+journals, lgtm!

File: srv/modules/runners/push.py (hunk 2 above)
Comment (swiftgist) [inline srv/modules/runners/push.py]: I thought this would make it stand out a bit.  Also, the logic is simple with prepending a string.

Comment (jschmid1) [inline srv/modules/runners/push.py]: Don't have a strong opinion on it. I just thought I'd share my initial thoughts. We can leave it as is.

**Comment by swiftgist** [comment]
Updated Makefile, spec file and added a log message.

**Pull request merged**